### ISSUE TITLE
Transactions by account includes unrelated Maybuy transactions

### ISSUE THREAD

**Issue by jacobjmarks** [up-banking/api#148]
Sequence of events:
1. A new 2Up Maybuy was created
2. The Maybuy was contributed to; completed; and the funds removed
   - Total transactions made: 5 (4 contributions and 1 transfer out)
4. The Maybuy was deleted
5. A new 2Up Saver was created
6. A single transaction was made on the new Saver

Now, when viewing this Saver on the mobile app, all is as expected, i.e. a single transaction can be seen.

However, when fetching transactions for this Saver via the `/accounts/:id/transactions` endpoint, a total of _6_ transactions can be seen. These include the single transaction that I made on the Saver (all I am expected to see), _as well as_ the 5 transactions that were made on the since-deleted Maybuy (and made before the Saver even existed).

In the response, all of these transactions share the same `relationships.account.data.id` value, that being the `data.id` of the new Saver account.

Edit: I have also since discovered that the `data.attributes.createdAt` value of the new 2Up Saver is dated _before_ I created it.

To summarize:

- 01/05: The _reported_ date that the 2Up Saver was created
  - I suspected that this is actually the date that the _Maybuy_ was created
- 05/06 - 17/06: Dates of Maybuy transactions
- 26/08: The _actual_ date that the 2Up Saver was created
- 26/08: 2Up Saver transaction was made

| Transaction/s | Visible on 2Up Saver account page | Visible on 2Up Saver transactions response
| - | - | - |
| 2Up Saver | Yes (Expected: Yes) | Yes (Expected: Yes)
| Maybuy | No (Expected: No) | **Yes (Expected: No)** |

**Comment by markbrown4** [comment]
I think you might be noticing the same thing as others have here
https://github.com/up-banking/api/issues/163#issuecomment-3894937104

Accounts are re-used across multiple savers.
This is surprising, but is accurate at the account level to what is really happening.
